code_fragments/SCALLOP_CAMS_GC_LA.do

```diff
@@ -0,0 +1,68 @@
+***This file lables all CAMS trips (including orphans) as limited Access or GC.
+
+
+ clear all
+
+jdbc connect , jar("$jar")  driverclass("$classname")  url("$nefscdb1_url")  user("$myuid") password("$nefscdb1pwd")
+ 
+jdbc load, exec ("select a.CAMSID, d.ACTIVITY_CODE_1, d.ACCESSAREA from CAMS_GARFO.CAMS_LAND a full outer join CAMS_GARFO.CAMS_subtrip d on a.CAMSID=d.CAMSID where a.ITIS_GROUP1 like '%SCALLOP%' and a.YEAR between $startyear and $endyear ") 
+ 
+save "${my_datadir}/intermediate/CAMS_SCALLOP_TRIPS.dta", replace 
+ 
+ 
+ ****load the orphans 
+  clear all
+
+jdbc connect , jar("$jar")  driverclass("$classname")  url("$nefscdb1_url")  user("$myuid") password("$nefscdb1pwd")
+ 
+jdbc load, exec ("select CAMSID, ACTIVITY_CODE_1, ACCESSAREA from CAMS_GARFO.CAMS_VTR_ORPHANS_SUBTRIP where MAIN_SPP_GRP like '%SCA%' and YEAR between $startyear and $endyear ") 
+ 
+ append using "${my_datadir}/intermediate/CAMS_SCALLOP_TRIPS.dta", replace 
+ 
+
+ split ACTIVITY_CODE_1, p(-)
+ 
+ gen activity_code = ACTIVITY_CODE_11 +"-"+ ACTIVITY_CODE_12
+ 
+ **label all Limited access trips 
+ gen LA= 1 if activity_code =="SES-SAA" |activity_code =="SES-SCA" 
+ replace LA = 0 if missing(LA)
+ 
+**label all general category trips 
+ gen GC= 1 if activity_code =="SES-SCG" 
+  replace GC = 0 if missing(GC)
+  
+ *** rename access area
+ 
+ rename ACCESSAREA access_area
+ replace access_area = "1" if access_area=="AA"
+  replace access_area = "0" if access_area!="1"
+  destring access_area, replace 
+  
+keep CAMSID access_area LA GC  
+ 
+**this is at the sub-trip level, so will have to make a call for whole trips. 
+***decision rule, if sub-trip is AA do whole trip as AA
+***decision rule if LA make whole trip LA
```

Comment:
I was a little surprised to see this code here. Are there many trips that are both AA and Not-AA?  I didn't think that was possible. If there are alot of trips, that means that either:
1. I'm wrong about the fishery.
2. There's something wrong with the way CAMS puts together data.
3. There's something about the way the CAMS data is put together that we are misunderstanding.

Reply:
I don't know. I am assuming something like this needs to be done as this is at the sub-trip level so there could be the possibility of two records and one being AA and one being not AA. Same with GC and LA. I am not sure it should happen but it probably happens that they differ for a single trip. This is used to merge to observer data for this work as a reminder so I need trip level info, not sub-trip